### PR TITLE
Fix supplemental policy rule options page

### PR DESCRIPTION
**Pull request by jgeurten**
Pull request changes the WDAC Wizard behavior surrounding supplemental policy creation. Specifically, 

1. Rule-options for supplemental policies. Bug #29385031. Only the following rules are allowed for supplemental policies now: 
- Audit mode
- Boot aduit on failure
- unsigned si policy
- invalidate eas on reboot
- disable flight signing
- runtime filepath rules

2. If the base policy the user is trying to supplement does not support supplemental policies, try to modify the base to allow supplemental policies. Bug #29380513

### CODE REVIEW

File: WDAC-Policy-Wizard/app/Resources/RulesDict.xml [hunk 1]
```diff
@@ -5,146 +5,167 @@
...
+    ValidSupplemental="True"
+	  ButtonMapping="button_13">
   </Rule>
   <Rule
     Name="ScriptEnforcement"
     AllowedValue="Disabled"
     Supported="True"
     RuleNumber="11"
-    ButtonMapping="button_2">
+    ValidSupplemental="False"
```
Comment (jsuther1974) [inline WDAC-Policy-Wizard/app/Resources/RulesDict.xml:96]: I assumed this **would** be configurable by a supplemental. Are you sure this is right?

File: WDAC-Policy-Wizard/app/Resources/RulesDict.xml [hunk 2]
```diff
@@ -5,146 +5,167 @@
...
+	  ButtonMapping="button_3">
   </Rule>
   <Rule
     Name="ManagedInstaller"
     AllowedValue="Enabled"
     Supported="True"
     RuleNumber="13"
-    ButtonMapping="button_7">
+    ValidSupplemental="False"
```
Comment (jsuther1974) [inline WDAC-Policy-Wizard/app/Resources/RulesDict.xml:112]: This is definitely wrong. ManagedInstaller is absolutely configurable via Supplemental.

File: WDAC-Policy-Wizard/app/Resources/RulesDict.xml [hunk 3]
```diff
@@ -5,146 +5,167 @@
...
+	  ButtonMapping="button_7">
   </Rule>
   <Rule
     Name="IntelligentSecurityGraphAuthorization"
     AllowedValue="Enabled"
     Supported="True"
     RuleNumber="14"
-    ButtonMapping="button_5">
+    ValidSupplemental="False"
```
Comment (jsuther1974) [inline WDAC-Policy-Wizard/app/Resources/RulesDict.xml:120]: This is wrong too. ISG is absolutely configurable via Supplemental as it expands what is allowed.

File: WDAC-Policy-Wizard/app/Resources/RulesDict.xml [hunk 4]
```diff
@@ -5,146 +5,167 @@
...
+	  ButtonMapping="button_5">
   </Rule>
   <Rule
     Name="InvalidateEAsonReboot"
     AllowedValue="Enabled"
     Supported="True"
     RuleNumber="15"
-    ButtonMapping="button_18">
+    ValidSupplemental="True"
```
Comment (jsuther1974) [inline WDAC-Policy-Wizard/app/Resources/RulesDict.xml:128]: Not sure this is accurate...

**Review by jsuther1974 — CHANGES_REQUESTED** [review]

File: WDAC-Policy-Wizard/app/Resources/RulesDict.xml [hunk 5]
```diff
@@ -5,146 +5,167 @@
...
+    ValidSupplemental="False"
+	  ButtonMapping="button_8">
   </Rule>
   <Rule
     Name="AuditMode"
     AllowedValue="Enabled"
     Supported="True"
     RuleNumber="3"
-    ButtonMapping="button_6">
+    ValidSupplemental="True"
```
[the remark this answers is not in the document]
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/Resources/RulesDict.xml]: Fixed in latest commit. Thanks for your help with these.

File: WDAC-Policy-Wizard/app/Resources/RulesDict.xml [hunk 6]
```diff
@@ -5,146 +5,167 @@
...
+	  ButtonMapping="button_6">
   </Rule>
   <Rule
     Name="FlightSigning"
     AllowedValue="Disabled"
     Supported="True"
     RuleNumber="4"
-    ButtonMapping="button_15">
+    ValidSupplemental="True"
```
[the remark this answers is not in the document]
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/Resources/RulesDict.xml]: Fixed in latest commit.

File: WDAC-Policy-Wizard/app/Resources/RulesDict.xml [hunk 7]
```diff
@@ -5,146 +5,167 @@
...
+	  ButtonMapping="button_4">
   </Rule>
   <Rule
     Name="UnsignedSystemIntegrityPolicy"
     AllowedValue="Enabled"
     Supported="True"
     RuleNumber="6"
-    ButtonMapping="button_10">
+    ValidSupplemental="True"
```
[the remark this answers is not in the document]
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/Resources/RulesDict.xml]: Fixed in latest commit.

File: WDAC-Policy-Wizard/app/Resources/RulesDict.xml [hunk 8]
```diff
@@ -5,146 +5,167 @@
...
+	  ButtonMapping="button_10">
   </Rule>
   <Rule
     Name="DebugPolicyAugmented"
     AllowedValue="Allowed"
     Supported="False"
     RuleNumber="7"
-    ButtonMapping="button_12">
+    ValidSupplemental="True"
```
[the remark this answers is not in the document]
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/Resources/RulesDict.xml:64]: Supported="False" will not paint the rule to the UI so the user cannot view/modify the rule. This is true for Rule #1,5 too

File: WDAC-Policy-Wizard/app/Resources/RulesDict.xml (hunk 3 above)
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/Resources/RulesDict.xml]: Fixed in latest commit.

File: WDAC-Policy-Wizard/app/Resources/RulesDict.xml (hunk 2 above)
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/Resources/RulesDict.xml]: Fixed in latest commit.

**Pull request merged**